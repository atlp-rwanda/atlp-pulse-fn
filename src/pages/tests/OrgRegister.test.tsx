/* eslint-disable */
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import Register from './../OrgRegister';
import { ApolloClient, ApolloProvider, InMemoryCache } from '@apollo/client';
import '@testing-library/jest-dom'

const client = new ApolloClient({ cache: new InMemoryCache() });

describe('Register an Organization', () => {
  test('Renders the register Page', () => {
    act(() => {
      render(
        <ApolloProvider client={client}>
          <MemoryRouter>
            <Register />
          </MemoryRouter>
        </ApolloProvider>,
      );
    });
    expect(screen.getByTestId('dataid')).toBeInTheDocument();
    expect(screen.getByTestId('input1')).toBeInTheDocument();
    act(() => {
      fireEvent.change(screen.getByTestId('input1'), {
        target: { value: 'Rwanda' },
      });
    });
  });
});
