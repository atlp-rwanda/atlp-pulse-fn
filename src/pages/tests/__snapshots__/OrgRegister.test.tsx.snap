// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register an Organization Renders the register Page 1`] = `
<div
  className="w-full min-h-screen -mt-3 -mb-16 text-left bg-gray-100 md:flex md:flex-col md:items-center md:justify-center xl:-mt-14 xl:-mb-28 dark:bg-dark-frame-bg sm:flex sm:flex-row sm:items-center sm:justify-center"
>
  <div
    className="bg-white sm:rounded-2xl py-8 md:py-0 md:shadow-2xl md:flex xl:w-3/4 md:w-[90%] md:max-w-4xl sm:w-full px-2 md:px-0 mx-20 md:mx-0 sm:shadow-none dark:shadow-2xl "
  >
    <div
      className="py-20 2xl:w-3/5 md:w-screen md:pl-14 lg:pl-32 xl:pl-24 md:py-8 sm:w-full sm:p-2 dark:bg-dark-bg dark:rounded-tl-xl dark:rounded-bl-xl"
    >
      <div
        className=""
      >
        <h2
          className="pb-6 ml-4 text-2xl font-bold md:ml-0 lg:text-3xl text-primary dark:text-dark-text-fill "
        >
          Register your organization
        </h2>
        <form
          onSubmit={[Function]}
        >
          <div
            className="flex flex-col items-left"
          >
            <label
              className="font-normal tracking-wide text-gray-700 dark:text-white"
            >
              Email
            </label>
            <div
              className="flex items-center w-full p-2 mb-6 border rounded border-primary dark:border-white md:w-3/4 dark:bg-dark-bg "
            >
              <svg
                className="mr-2 text-gray-400"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M464 64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V112c0-26.51-21.49-48-48-48zm0 48v40.805c-22.422 18.259-58.168 46.651-134.587 106.49-16.841 13.247-50.201 45.072-73.413 44.701-23.208.375-56.579-31.459-73.413-44.701C106.18 199.465 70.425 171.067 48 152.805V112h416zM48 400V214.398c22.914 18.251 55.409 43.862 104.938 82.646 21.857 17.205 60.134 55.186 103.062 54.955 42.717.231 80.509-37.199 103.053-54.947 49.528-38.783 82.032-64.401 104.947-82.653V400H48z"
                />
              </svg>
              <input
                className="flex-1 text-sm bg-white outline-none dark:border-white dark:bg-dark-bg dark:text-white"
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
              />
            </div>
            <div
              className="-mt-6"
            />
            <label
              className="mt-3 mb-2 font-normal tracking-wide text-gray-700 dark:text-white"
            >
              Organization name
            </label>
            <div
              className="flex items-center w-full p-2 mb-6 border rounded border-primary dark:border-white md:w-3/4 dark:bg-dark-bg "
            >
              <svg
                className="mr-2 text-gray-400"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M313.6 304c-28.7 0-42.5 16-89.6 16-47.1 0-60.8-16-89.6-16C60.2 304 0 364.2 0 438.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-25.6c0-74.2-60.2-134.4-134.4-134.4zM400 464H48v-25.6c0-47.6 38.8-86.4 86.4-86.4 14.6 0 38.3 16 89.6 16 51.7 0 74.9-16 89.6-16 47.6 0 86.4 38.8 86.4 86.4V464zM224 288c79.5 0 144-64.5 144-144S303.5 0 224 0 80 64.5 80 144s64.5 144 144 144zm0-240c52.9 0 96 43.1 96 96s-43.1 96-96 96-96-43.1-96-96 43.1-96 96-96z"
                />
              </svg>
              <input
                className="flex-1 text-sm bg-white outline-none dark:border-white dark:bg-dark-bg dark:text-white "
                name="name"
                onBlur={[Function]}
                onChange={[Function]}
                type="text"
              />
            </div>
            <div
              className="-mt-6"
            />
            <label
              className="mt-3 mb-2 font-normal tracking-wide text-gray-700 dark:text-white"
            >
              Organization Description
            </label>
            <div
              className="flex items-center w-full p-2 mb-3 border rounded border-primary dark:border-white md:w-3/4 dark:bg-dark-bg "
            >
              <svg
                className="mr-2 -mt-5 text-gray-400"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M290.74 93.24l128.02 128.02-277.99 277.99-114.14 12.6C11.35 513.54-1.56 500.62.14 485.34l12.7-114.22 277.9-277.88zm207.2-19.06l-60.11-60.11c-18.75-18.75-49.16-18.75-67.91 0l-56.55 56.55 128.02 128.02 56.55-56.55c18.75-18.76 18.75-49.16 0-67.91z"
                />
              </svg>
              <textarea
                className="flex-1 text-sm bg-white outline-none dark:border-white dark:bg-dark-bg dark:text-white"
                name="description"
                onBlur={[Function]}
                onChange={[Function]}
              />
            </div>
            <div
              className="-mt-4"
            />
            <div
              className="flex w-80  md:w-[75%] justify-between rounded mb-5 mt-5"
            >
              <label
                className="flex text-xs items-left dark:text-dark-text-fill "
              >
                <input
                  className="mr-1 dark:text-dark-text-fill dark:border-white dark:bg-dark-frame-bg"
                  name="remember"
                  type="checkbox"
                />
                I agree to the
                <span
                  className="mx-1 text-primary"
                >
                  Terms & Conditions of
                   
                </span>
                 
                <span>
                  DevPulse
                </span>
              </label>
            </div>
            <button
              className="w-1/4 text-center mx-[20%] py-2 rounded-lg bg-primary text-white hover:bg-green-dark focus:outline-none"
              type="submit"
            >
              Register
            </button>
          </div>
        </form>
      </div>
    </div>
    <div
      className="text-white md:w-2/5 bg-primary rounded-tr-2xl rounded-br-2xl dark:rounded-br-2xl dark:rounded-tr-2xl md:py-4 md:px-2 sm:w-full sm:px-12 sm:py-4 md:block sm:hidden dark:text-dark-text-fill "
    >
      <h2
        className="text-center 2xl:text-xl md:text-xl md:font-bold md:pb-6"
      >
        Get Started
      </h2>
      <h2
        className="px-2 pb-4 font-sans text-lg font-bold text-center text-dark-text-fill"
      >
        Registration_page_heading
      </h2>
      <p
        className="max-w-[600px] text-left text-white pl-5 pb-6 text-md"
      >
        Registration_page_paragraph
        <a
          className="ml-4 font-bold text-white border-b-1 pr-52"
          href="/get-started"
          onClick={[Function]}
        >
          Explore
        </a>
      </p>
      <h3
        className="px-5 mb-2 text-lg font-bold text-left text-white"
      >
        Your organization's link
      </h3>
      <button
        className="px-3 py-2 mt-3 ml-20 font-bold bg-white rounded text-primary"
        type="button"
      >
        <span>
          .pulse.org
        </span>
      </button>
    </div>
  </div>
</div>
`;
